### PR TITLE
[stdapi] support_unicode_characters_for_execute Fix #14428

### PR DESCRIPTION
**Pull request by cn-kali-team**
Unicode characters cannot appear in the executable file path
Please check [bug](https://github.com/rapid7/metasploit-framework/issues/14428)

## Expected behavior
```
meterpreter > ls                       
Listing: C:\Users\Administrator\Desktop
40777/rwxrwxrwx   4096      dir   2020-10-29 09:29:49 +0800  新建文件夹  
100777/rwxrwxrwx  1074688   fil   2020-11-24 11:46:46 +0800  新建文件夹.exe 
                                                                                                                                                   
meterpreter > execute -f 新建文件夹.exe 
Process 4740 created.
```

## Current behavior

```
meterpreter > ls                       
Listing: C:\Users\Administrator\Desktop
40777/rwxrwxrwx   4096      dir   2020-10-29 09:29:49 +0800  新建文件夹  
100777/rwxrwxrwx  1074688   fil   2020-11-24 11:46:46 +0800  新建文件夹.exe 
                                                                                                                                                   
meterpreter > execute -f 新建文件夹.exe 
[-] 1069: Operation failed: The system cannot find the file specified.
```

### CODE REVIEW

File: c/meterpreter/source/extensions/stdapi/server/sys/process/process.c [hunk 1]
```diff
@@ -381,14 +381,15 @@ DWORD request_sys_process_execute(Remote *remote, Packet *packet)
...
+			wchar_t *commandLine_w = met_api->string.utf8_to_wchar(commandLine);
+			if (!CreateProcessAsUserW(pToken, NULL, commandLine_w, NULL, NULL, inherit, createFlags, pEnvironment, NULL, (LPSTARTUPINFOW)&si, &pi))
 			{
 				LPCREATEPROCESSWITHTOKENW pCreateProcessWithTokenW = NULL;
 				HANDLE hAdvapi32 = NULL;
 				wchar_t * wcmdline = NULL;
 				wchar_t * wdesktop = NULL;
 				size_t size = 0;
-
+				free(commandLine_w);
```
Comment (timwr) [inline c/meterpreter/source/extensions/stdapi/server/sys/process/process.c:392]: I think we also need to free(commandLine_w) if CreateProcessAsUserW fails. Let me try send a pull request that fixes this.

**Comment by timwr** [comment]
@cn-kali-team please see https://github.com/cn-kali-team/metasploit-payloads/pull/2
It should fix the bug, but it also cleans up the code a little bit and ensures we don't leak any memory on certain failures.
If you're happy with it let me know and I'll merge it.

**Comment by cn-kali-team** [comment]
> @cn-kali-team please see [cn-kali-team#2](https://github.com/cn-kali-team/metasploit-payloads/pull/2)
> It should fix the bug, but it also cleans up the code a little bit and ensures we don't leak any memory on certain failures.
> If you're happy with it let me know and I'll merge it.

Thank you very much for your work

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

**Comment by timwr** [comment]
Fixed with https://github.com/rapid7/metasploit-payloads/commit/c15ae41e8c95b89d200a792ccea79246e8c58a68
Closing manually because I didn't pull in the merge commit.

**Pull request closed without merging**